calico_containers/calico_ctl/node.py

```diff
@@ -536,24 +548,33 @@ def warn_if_hostname_conflict(ip):
     """
     # If there's already a calico-node container on this host, they're probably
     # just re-running node to update one of the ip addresses, so skip..
-    if not docker_client.containers(filters={'name': 'calico-node'}):
-        # Otherwise, check if another host with the same hostname
-        # is already configured
-        try:
-            current_ipv4, _ = client.get_host_bgp_ips(hostname)
-        except KeyError:
-            # No other machine has registered configuration under this hostname.
-            # This must be a new host with a unique hostname, which is the
-            # expected behavior.
-            pass
-        else:
-            if current_ipv4 != "" and current_ipv4 != ip:
+    try:
+        current_ipv4, _ = client.get_host_bgp_ips(hostname)
+    except KeyError:
+        # No other machine has registered configuration under this hostname.
+        # This must be a new host with a unique hostname, which is the
+        # expected behavior.
+        pass
+    else:
+        if current_ipv4 != "" and current_ipv4 != ip:
+            try:
+                if not docker_client.containers(filters={'name': 'calico-node'}):
+                    # Calico-node isn't running on this host.
+                    # There may be another host using this hostname.
+                    print_paragraph("WARNING: Hostname '%s' is already in use "
+                            "with IP address %s. Calico requires each compute host to "
+                            "have a unique hostname. If this is your first time "
+                            "running 'calicoctl node' on this host, ensure that "
+                            "another host is not already using the "
+                            "same hostname." % (hostname, ip))
+            except IOError:
+                # Couldn't connect to docker to confirm calico-node is running.
```

Comment:
Ideally we'd detect if calico-felix is running on the system here, and if it is not, then drop the warning. Instead of designing a way to detect if Felix is running or not, I figured it'd be safer to always warn for now.